src/librustdoc/html/static/main.js

```diff
@@ -1962,6 +1964,21 @@ function defocusSearchBar() {
         }
     };
 
+    /// Render HTML as plaintext:
+    ///
+    /// * Replace "<code>foo</code>" with "`code`"
+    /// * Strip all other HTML tags
+    ///
+    /// Note that it just uses a regex to strip the other HTML tags,
+    /// so it may not work correctly for cases where there is an angle
+    /// bracket inside quotes.
+    ///
+    /// Used by the dynamic sidebar crate list renderer.
+    function renderHTMLAsPlaintext(html) {
+        return html.replace('<code>', '`').replace('</code>', '`')
+                   .replace(/<[^>]+>/, '');
+    }
```

Comment:
This seems really fragile ... is there a way to do this in rustdoc itself, where we know what was added by rustdoc and what was added by the user? The issue is that anything we store makes the index take longer to render ... I'm not sure the best approach here.

@GuillaumeGomez do you have ideas?

Reply:
It's definitely not the right approach. The good approach is to generate the HTML directly from rustdoc into the search index.

Reply:
@GuillaumeGomez What do you mean? The HTML *is* being rendered into the search index; that's why we need this function.

Reply:
What I mean is that if there is a `code` attribute in the first line, the transformation into HTML should be done in rustdoc when generating the doc. Nothing should be done on the JS side.

Reply:
It is being transformed into HTML in rustdoc. The JS transforms the HTML to plaintext for the alt-text.

Reply:
Oh I see! Sorry, I was completely out of it. But then, is it really worth it?

Reply:
That's okay :)

Is what worth it?

Reply:
To generate the alt-text with this information. Isn't `DOMElement.innerText` enough? Do we really want to keep things like backquote?

Reply:
I still don't understand what you mean :/

The JS code previously used the search index to fill the crate list in the sidebar, and used the (plaintext) summary for search result as the alt-text (actually the `title` attribute) on each crate in the list. But now the search result summaries are HTML. However, we can't use HTML in the alt-text since it is rendered literally to the user, i.e. if you hover over an element it will show a tooltip that looks like "This crate has a \<code\>Foo\</code\> struct". It used to look like "This crate has a \`Foo\` struct".

So the solution I went with for now is to still get the alt-text from the search index, but to first pass it through a function to convert it to plaintext. So I think the worst that can happen if this function doesn't work correctly is that the alt-text will look a little funny. Personally I think for now we should go with this approach and then in a separate PR change it so that it doesn't convert the HTML to plaintext in the JS. That way this PR's scope will stay somewhat small.

**P.S.** When I say "alt-text" I mean the text in the tooltip that is shown when you hover over an element.

Reply:
So I understood that part correctly. What I mean is that what you're doing is just way too much. :) First: never parse HTML directly, if we really want to do that, then use the DOM api (`childNodes` and equivalents). Second: this is way too much, just get the text using the text you get after setting the HTML to the DOMElement and then use `innerText` which will return the text only and no HTML.

Reply:
I think I may understand what you mean: You're suggesting that I use something https://stackoverflow.com/a/21870431 and use the built-in `DOMParser` to parse the HTML and then just use `.innerText` on it? I hadn't thought of that; good idea!

Reply:
I just realized that we created the whole table results as a string that we set to the `innerHTML`. That's why what I said didn't make sense (sorry!). Instead of this whole thing, we could instead create each line using `document.createElement` and so on. And then, on the created table entry (the `<tr>`), you can then get its last `<td>.innerText`. No need to parse anything. But you might need to rewrite a bit how we generate the results output. I can do it if you want?

Reply:
I'm sorry, but I still don't understand what you are suggesting. We could just use `DOMParser` to parse the HTML and then use `.innerText`, but then inline code would not have backticks around it. But that's not a big deal since this is just alt-text. I don't see how we can have alt-text without stripping the tags of the HTML.

Reply:
This parsing is *purely* for the alt-text (`title` attribute), not for what's displayed in the rows. I'm not sure why what you're suggesting is relevant to rendering the alt-text. In a different PR, it may make sense to refactor this code, but I don't see why it's necessary now.

Reply:
I'll send a PR to make the change I suggested then, so that then you can use it. It'll be simpler haha.

Reply:
I opened https://github.com/rust-lang/rust/pull/78805. You can cherry-pick the commit if you want or just land the PR and then rebase on it, as you prefer.

Reply:
Oh, that code is what I thought you meant :)

Reply:
I ended up using something similar to what you suggested, but instead used `DOMParser`, which is designed explicitly for parsing HTML and manipulating it as a DOM element. It also seems to be supported in pretty much every browser.